src/utils/search/search.resolver.ts

```diff
@@ -0,0 +1,28 @@
+import { UseGuards } from '@nestjs/common';
+import { Args, Resolver, Query } from '@nestjs/graphql';
+import { Roles } from '../decorators/roles.decorator';
+import { GqlAuthGuard } from '../authentication/graphql.guard';
+import { SearchService } from './search.service';
+import { ISearchResult } from './search-result.interface';
+import { Profiling } from '../logging/logging.profiling.decorator';
+import { SearchInput } from './search-input.dto';
+import { SearchResult } from './search-result.dto';
+import { RestrictedGroupNames } from '../../domain/user-group/user-group.entity';
+
+@Resolver()
+export class SearchResolver {
+  constructor(private searchService: SearchService) {}
+
+  @Roles(RestrictedGroupNames.Members)
```

Comment:
Currently, we don't have a way to set a 'minimum' role so you need to add community-admins and ecoverse-admins here as well.